ui-tests/src/test/java/org/openmrs/reference/DuplicatePatientRegisterTest.java

```diff
@@ -1,74 +1,75 @@
 
+import org.hamcrest.Matchers;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.openmrs.reference.groups.BuildTests;
 import org.openmrs.reference.helper.PatientGenerator;
 import org.openmrs.reference.helper.TestPatient;
-import org.openmrs.reference.page.HeaderPage;
-import org.openmrs.reference.page.HomePage;
+import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
 import org.openmrs.reference.page.RegistrationPage;
-import org.openmrs.uitestframework.test.TestBase;
 
 import java.awt.*;
 
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertThat;
 
 
-public class DuplicatePatientRegisterTest  extends TestBase {
-    private HeaderPage headerPage;
-    private RegistrationPage registrationPage;
-    private HomePage homePage;
-    private TestPatient patient1;
+public class DuplicatePatientRegisterTest  extends ReferenceApplicationTestBase {
+    private TestPatient patient;
 
     @Before
     public void setUp() throws AWTException {
-        headerPage = new HeaderPage(driver);
-        patient1 = PatientGenerator.generateTestPatient();
-
-        assertPage(page);
-        
-        homePage = new HomePage(page);
-        registrationPage = new RegistrationPage(page);
-        assertPage(homePage);
+        patient = PatientGenerator.generateTestPatient();
     }
 
-    private void registerAPatient(TestPatient patient) throws InterruptedException {
-
-        homePage.goToRegisterPatientApp();
-        patient.familyName = "Smith";
-        patient.middleName = "";
-        patient.givenName = "Bob";
-        patient.gender = "Male";
-        patient.birthYear = "1999";
-        patient.birthMonth = "January";
-        patient.birthDay = "1";
-        patient.address1 = "address";
-        patient.address2 = "address";
+    // Test for RA-714
+    @Test
+    @Category(BuildTests.class)
+    public void duplicateRegisterTest() throws InterruptedException {
+        RegistrationPage registrationPage = homePage.goToRegisterPatientApp();
         registrationPage.enterPatient(patient);
+        ClinicianFacingPatientDashboardPage dashboardPage = registrationPage.confirmPatient();
+
+        patient.uuid = dashboardPage.getPatientUuidFromUrl();
+
+        dashboardPage.goToHomePage();
+
+        registrationPage = homePage.goToRegisterPatientApp();
+        registrationPage.enterPatient(patient);
+
+        String name = registrationPage.getSimilarPatientName();
+        assertThat(name, Matchers.is(patient.givenName + " " + patient.familyName));
+
+        String birthDate;
+        if(patient.birthDay.length() == 1) {
+            birthDate = "0" + patient.birthDay + "." + patient.birthMonth.substring(0,3) + "." + patient.birthYear;
+        }
+        else {
+            birthDate = patient.birthDay + "." + patient.birthMonth.substring(0,3) + "." + patient.birthYear;
+        }
```

Comment:
I thought it will work fine like that and didn' t have to change much. Do I have to change this?